addons/mail/static/src/models/thread/thread.js

```diff
@@ -1016,15 +1016,15 @@ function factory(dependencies) {
             // serverMessageUnreadCounter
             if (this.lastSeenByCurrentPartnerMessageId < firstMessage.id) {
                 const fetchedNotSeenMessages = this.orderedMessages.filter(message =>
-                    message.id > this.serverLastMessageId
+                    message.id > this.serverLastMessageId && message.author !== this.messagingCurrentPartner
```

Comment:
Would be best if `serverLastMessageId` had the correct value instead (the id of the message being just posted for its author).